llvm/lib/Target/RISCV/MCTargetDesc/RISCVMatInt.cpp

```diff
@@ -436,6 +437,46 @@ InstSeq generateInstSeq(int64_t Val, const MCSubtargetInfo &STI) {
   return Res;
 }
 
+SmallVector<MCInst, 8>
+generateMCInstSeq(int64_t Val, const MCSubtargetInfo &STI, MCRegister DestReg) {
```

Comment:
I think SmallVectors are typically "returned" by a reference argument to avoid a copy

```suggestion
void
generateMCInstSeq(int64_t Val, const MCSubtargetInfo &STI, MCRegister DestReg, SmallVectorImpl<MCInst> &Insts) {
```

And you can use SmallVectorImpl so you don't hardcode the small size: https://www.llvm.org/docs/ProgrammersManual.html#id47

Reply:
> I think SmallVectors are typically "returned" by a reference argument to avoid a copy

"named return value optimization" or "guaranteed copy elision" should avoid a copy when returning a SmallVector local variable.